lib/cors_plug.ex

```diff
@@ -16,17 +16,19 @@ defmodule CORSPlug do
   end
 
   def init(options) do
-    Dict.merge defaults, options
+    Keyword.merge(defaults, options)
   end
 
   def call(conn, options) do
-    conn = put_in conn.resp_headers, conn.resp_headers ++ headers(conn, options)
+    new_headers = headers(conn, options)
+    conn = put_in(conn.resp_headers, new_headers)
     case conn.method do
-      "OPTIONS" -> halt send_resp conn, 204, ""
+      "OPTIONS" -> conn |> send_resp(204, "") |> halt()
       _method   -> conn
     end
   end
 
+  # headers specific to OPTIONS request
```

Comment:
Thanks for adding documentation, much needed. May I ask you to use the official `@doc` notation, please?